packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -4283,21 +4289,46 @@ export class ContainerRuntime
 				} = {};
 				const trace = Trace.start();
 
-				const versions = await this.storage.getVersions(
-					null,
-					1,
-					"prefetchLatestSummaryBeforeClose",
-					FetchSource.noCache,
-				);
-				assert(!!versions && !!versions[0], 0x137 /* "Failed to get version from storage" */);
-				stats.getVersionDuration = trace.trace().duration;
+				let maybeSnapshotTree: ISnapshotTree | null;
+				const scenarioName = "prefetchLatestSummaryBeforeClose";
+				// If loader supplied us the ISnapshot when loading, the new getSnapshotApi is supported and feature gate is ON, then use the
+				// new API, otherwise it will reduce the service performance because the service will need to recalculate the full snapshot
+				// in case previously getSnapshotApi was used and now we use the getVersions API.
+				if (
+					this.isSnapshotInstanceOfISnapshot &&
+					this.storage.getSnapshot !== undefined &&
+					this.mc.config.getBoolean("Fluid.Container.UseLoadingGroupIdForSnapshotFetch2") ===
+						true
+				) {
+					const snapshot = await this.storage.getSnapshot({
+						scenarioName,
+						fetchSource: FetchSource.noCache,
```

Comment:
I wonder if it would be useful to pre-add groupIds, or is that an over optimization here?

Reply:
Since this is a snapshot which we will load from, I want to keep that part only (for starters) and later maybe we can do that optimization :)